test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,607 @@
+
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+            const timerSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'setSessionTimer'
+            );
+
+            // Session Manager relies on persistence to determine last time seen (LES)
+            // Also requires sid to verify session exists
+            sinon.stub(mpInstance._Persistence, 'getPersistence').returns({
+                gs: {
+                    les: twentyMinutesAgo,
+                    sid: 'fake-session-id',
+                },
+            });
+
+            mpInstance._SessionManager.endSession();
+
+            expect(mpInstance._Store.sessionId).to.equal('fake-session-id');
+
+            // When session is not timed out, setSessionTimer is called to keep track
+            // of current session timeout
+            expect(timerSpy.getCalls().length).to.equal(1);
+        });
+
+        it('should end session if session times out', () => {
+            const now = new Date();
+            const hourAgo = new Date();
+            hourAgo.setMinutes(now.getMinutes() - 60);
+
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+            const persistenceSpy = sinon.spy(mpInstance._Persistence, 'update');
+
+            // Session Manager relies on persistence to determine last time seen (LES)
+            // Also requires sid to verify session exists
+            sinon.stub(mpInstance._Persistence, 'getPersistence').returns({
+                gs: {
+                    les: hourAgo,
+                    sid: 'fake-session-id',
+                },
+            });
+
+            mpInstance._SessionManager.endSession(true);
+
+            expect(mpInstance._Store.sessionId).to.equal(null);
+            expect(mpInstance._Store.dateLastEventSent).to.equal(null);
+            expect(mpInstance._Store.sessionAttributes).to.eql({});
+
+            // Persistence isn't necessary for this feature, but we should test
+            // to see that it is called in case this ever needs to be refactored
+            expect(persistenceSpy.called).to.equal(true);
+        });
+    });
+
+    describe('#setSessionTimer', () => {
+        it('should end a session after the timeout expires', done => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+            const endSessionSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'endSession'
+            );
+
+            // Start Timer
+            mpInstance._SessionManager.setSessionTimer();
+
+            // Progress 29 minutes to make sure end session has not fired
+            clock.tick(29 * 60000);
+            expect(endSessionSpy.called).to.equal(false);
+
+            // Progress one minutes to make sure end session fires
+            clock.tick(30 * 60000);
```

Comment:
This isn't progressing one more minute, it's progressing 30 more minutes.  Still works, but the comment is off, or the code is off.
```suggestion
            clock.tick(1 * 60000);
```